### PR TITLE
Add Datadog sample for NSB v8

### PR DESCRIPTION
**Pull request by timbussmann**
needs some refactoring since the `Feature` isn't required anymore to get to the metrics options.

### CODE REVIEW

File: samples/logging/datadog/Core_8/Endpoint/LoadSimulator.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using NServiceBus;
+
+// Simulates busy (almost no delay) / quite time in a sine wave
```
Comment (mikeminutillo) [inline samples/logging/datadog/Core_8/Endpoint/LoadSimulator.cs:6]:
```suggestion
// Simulates busy (almost no delay) / quiet time in a sine wave
```

File: samples/logging/datadog/sample.md [hunk 2]
```diff
@@ -38,7 +38,7 @@ Setup Datadog client. Let' s assume the Datadog agent runs on default address an
 
 snippet: setup-datadog-client
 
-Custom observers need to be registered for the metric probes provided via `NServiceBus.Metrics`. This is configured in the `DatadogFeature`. The registered observers convert NServiceBus.Metric *Signals* to Datadog *Count* and NServiceBus.Metric *Durations* to Datadog *Timer* 
+Custom observers need to be registered for the metric probes provided via the `NServiceBus.Metrics` package. The registered observers convert NServiceBus.Metric *Signals* to Datadog *Count* and NServiceBus.Metric *Durations* to Datadog *Timer*.
```
Comment (mikeminutillo) [inline samples/logging/datadog/sample.md:41]: The v7 version of the sample still uses the feature. Is it worth backporting this change (or making the v8 sample still a feature)?

**Review by mikeminutillo — APPROVED** [review]
I think the idea of having it in a feature was so that it was encapsulated and re-usable, not because it was necessary. It does add noise to the test though so maybe it's worth dropping it. It still had hard-coded connection strings inside the feature and it might detract from the sample to show how to set up a configuration extension method to enable the feature and pass connection details around.

File: samples/logging/datadog/sample.md (hunk 2 above)
Comment (timbussmann) [reply samples/logging/datadog/sample.md:41]: I don't see a problem with v7 using a feature, the text doesn't go deeper so I don't think this is a confusing change.

But we can't use a feature unless we take some extra effort to pass everything through the settings to set up the observes in the feature instead. I think we should not use a feature. If we really want, we can encapsulate the setup in a dedicated class but I didn't see that much value in that.

File: samples/logging/datadog/Core_8/Endpoint/Program.cs [hunk 3]
```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using NServiceBus;
+using StatsdClient;
+
+class Program
+{
+    const string EndpointName = "Samples.Metrics.Tracing.Endpoint";
```
Comment (timbussmann) [inline samples/logging/datadog/Core_8/Endpoint/Program.cs:10]:
```suggestion
    const string EndpointName = "Samples.Metrics.Tracing.Endpoint";

```

File: samples/logging/datadog/sample.md (hunk 2 above)
Comment (mikeminutillo) [reply samples/logging/datadog/sample.md:41]: Fair enough. The feature isn't the point of the sample and people are unlikely to download both versions and compare them, wondering why they're different.

**Comment by timbussmann** [comment]
so btw. in the new relic sample (https://github.com/Particular/docs.particular.net/pull/5672) I've extracted the setup into a dedicated class. Is that making the sample easier to read?

**Comment by mikeminutillo** [comment]
> so btw. in the new relic sample (#5672) I've extracted the setup into a dedicated class. Is that making the sample easier to read?

I think it does. It means that someone who downloads the code will find everything needed for integration in one place that is separate from everything else.

**Pull request merged**